src/main/java/selic/re/discordbridge/DiscordBot.java

```diff
@@ -1,492 +1,45 @@
-            return new LiteralText(message);
-        }
-
-        GameMessageConverter.Results results = convertGameMessage(message, chatChannel, discord);
-        sendChatMessage(author, results.discordOutput);
-        return results.gameOutput;
-    }
-
-    public void sendChatMessage(GameProfile player, String msg) {
-        Member member = playerLookup.getDiscordMember(getGuild(), player);
-        String name = member != null ? member.getEffectiveName() : player.getName();
-
-        if (webhook != null) {
-            webhook.send(startWebhook(player.getId(), name).setContent(msg).build());
-        } else {
-            TextChannel chatChannel = discord.getTextChannelById(config.channelId);
-            if (chatChannel != null) {
-                chatChannel.sendMessage("<" + name + "> " + msg).queue();
-            }
-        }
-    }
-
-    public void sendSystemMessage(Text text) {
-        if (webhook != null) {
-            webhook.send(new WebhookMessageBuilder().setContent(minecraftToDiscord(text)).build());
-        } else {
-            TextChannel chatChannel = discord.getTextChannelById(config.channelId);
-            if (chatChannel != null) {
-                chatChannel.sendMessage(minecraftToDiscord(text)).queue();
-            }
-        }
-    }
-
-    public Text getDiscordName(PlayerEntity player) {
-        Member member = playerLookup.getDiscordMember(guild, new GameProfile(player.getUuid(), null));
-        if (member != null) {
-            return DiscordFormattingConverter.discordUserToMinecraft(member.getUser(), guild, false);
-        }
-        return null;
-    }
-
-    protected WebhookMessageBuilder startWebhook(UUID id, String name) {
-        String avatar = config.getAvatarUrl(id);
-
-        return new WebhookMessageBuilder().setUsername(name).setAvatarUrl(avatar);
-    }
-
-    private void updateChannels() {
-        nextTopicUpdateTime = Instant.now().plus(TIME_BETWEEN_TOPIC_UPDATES);
-        topicNeedsUpdating = false;
-
-        TextChannel chatChannel = discord.getTextChannelById(config.channelId);
-
-        if (chatChannel != null && config.updateTopic) {
-            String topic = config.getTopicName(getPlayerNames());
-            chatChannel.getManager().setTopic(topic).queue(
-                null,
-                new ErrorHandler().handle(ErrorResponse.MISSING_PERMISSIONS, c -> {
-                    LogManager.getLogger().warn("Missing permissions to change channel info!");
-                    // Don't try again
-                    config.updateTopic = false;
-                })
-            );
-        }
-
-        GuildChannel renameChannel = discord.getGuildChannelById(config.renameChannelId);
-        int playerCount = server.getCurrentPlayerCount();
-
-        if (renameChannel != null) {
-            renameChannel.getManager().setName(config.getRenameChannelName(playerCount)).queue();
+public interface DiscordBot {
+    static DiscordBot instance() {
```

Comment:
This strikes me as rather odd. Shouldn't you be the one managing the static instance, instead of there being N static instances per backend? I would certainly move the initialization into constructors and manage the static instance within the `DiscordBot` class itself, whether it is `UninitializedDiscordBot` (maybe `UninitializedBackend` would be better?) or `DiscordBotImpl`